contracts/smart-wallet/utils/AccountExtension.sol

```diff
@@ -25,18 +23,13 @@ import "../interfaces/IAccountFactory.sol";
 //   \$$$$  |$$ |  $$ |$$ |$$ |      \$$$$$$$ |\$$$$$\$$$$  |\$$$$$$$\ $$$$$$$  |
 //    \____/ \__|  \__|\__|\__|       \_______| \_____\____/  \_______|\_______/
 
-contract AccountExtension is ContractMetadata, PermissionsEnumerable, ERC721Holder, ERC1155Holder {
+contract AccountExtension is ContractMetadata, AccountPermissions, ERC721Holder, ERC1155Holder {
```

Comment:
how come both `AccountCore` and `AccountExtension` both inherit `AccountPermissions` ? would have expected it only in the core

Reply:
Both require the functionality of `AccountPermissions` since both contracts have logic that works with permission checks. 

The `AccountPermissions` contract uses the library storage layout, so even though the two seperate contracts inherit it, the same storage slots are read/written to in a dynamic contract using both `AccountCore` + `AccountExtension`.